HeatStroke/GOComponents/GameObject.cpp

```diff
@@ -113,6 +120,66 @@ void GameObject::DeleteAllComponents()
 	m_mComponentMap.clear();
 }
 
+
+void GameObject::AddChild(GameObject* p_pChild)
+{
+	if (p_pChild != nullptr)
+	{
+		m_mChildMap.insert(std::pair<std::string, GameObject*>(p_pChild->GetGUID(), p_pChild));
+		p_pChild->SetParent(this);
+		m_Transform.AddChild(&(p_pChild->GetTransform()));
```

Comment:
@David-Hanna The hierarchical transform doesn't seem to be working properly.  Not sure if it's an unknown bug with `HierarchicalTransform` or if I'm doing something wrong.

Running the game and checking the console will show the error.  Essentially, it seems like the children are just getting the transform of their parent, not a combination of their own and their parent.  Translate seems to be fixed if you call `SetTranslate(x, y, z)` on the parent.  This doesn't seem to work for scale.